hooks/src/index.js

```diff
@@ -30,17 +30,17 @@ const RAF_TIMEOUT = 100;
 let prevRaf;
 
 options._diff = (internal, vnode) => {
-	currentComponent = null;
+	currentInternal = null;
 	if (oldBeforeDiff) oldBeforeDiff(internal, vnode);
 };
 
 options._render = internal => {
 	if (oldBeforeRender) oldBeforeRender(internal);
 
-	currentComponent = internal._component;
+	currentInternal = internal;
 	currentIndex = 0;
 
-	const hooks = currentComponent.__hooks;
+	const hooks = currentInternal.__hooks;
```

Comment:
I'm thinking that `__hooks` should still live off of the `_component` property. Perhaps a better name for that property would be `_componentData` or `_instance`.

The idea is that Internal's shape should be pre-defined up front and not change to keep our core algo fast and avoid re-compiles on shape changes. Since at this point class components are still in core, `_component` will already exist and be an object but in the future if that's not true, I'd imagine in `options.render`, hooks would do something like `if (internal._component == null) { internal._component = { __hooks: [], ... /* etc. */ }`

Reply:
So it was good before? Figured we'd want tobget rid of the property all together 😅

Reply:
Yeah I like it either being on _component, or on that new _data "grab bag" object.

Reply:
personally would prefer moving away from `_component`, is `_data` already started if so would it be a good keep-sake for all `hooks`/`components`/`composition`?